image/templates/helm/stackrox-central/templates/01-central-00-serviceaccount.yaml

```diff
@@ -8,9 +8,11 @@ metadata:
   labels:
     {{- include "srox.labels" (list . "serviceaccount" "central") | nindent 4 }}
   annotations:
-    {{- include "srox.annotations" (list . "serviceaccount" "central") | nindent 4 }}
+    {{- if and (eq ._rox.env.openshift 4) (not (hasKey ._rox.customize.central.envVars "ROX_MANAGED_CENTRAL" )) }}
```

Comment:
I don't think this is the right approach. I know why you've chosen it instead of adding a control knob at the level of Helm values, or rather, I can imagine two reasons:
- because you don't want to expose managed services mode as an "official" setting to customers
- because we are limited by which values can be set through the operator, and we certainly don't want to add a `managedServices` field to our Central CRD.

Imho, the first reason is a non-reason. Nobody says such a Helm chart value would need to be documented, and I don't think an undocumented value that is checked somewhere in the template via `{{- if ._rox._managedService }}` or so is any more exposed than the "magical" env var which is also referenced here in the template.

The second reason is understandable, but the implementation here won't work. As you can see in https://github.com/stackrox/stackrox/blob/master/operator/pkg/central/values/translation/translation.go#L74 in conjunction with https://github.com/stackrox/stackrox/blob/fa4592ee07ad8be1f8664edb66c87bb7324c74c3/operator/pkg/values/translation/translation.go#L37 , the operator only supports setting env vars _globally_, for all components. That is fine for the part inside central which reads the setting via `os.Getenv`, but it means that in the Helm chart rendering done by the operator, with values set from the translation logic linked above, the setting here is actually called `._rox.customize.envVars`.

Changing `._rox.customize.central.envVars` to `._rox.customize.envVars` is certainly a quick fix that should get the job done, and it might be the right choice depending on the time-sensitivity. I would argue that it's not the right approach because it's a leaky abstraction - we write the Helm chart based on some detail knowledge about how the operator implementation works. Meh.

One idea I have off the top of my head on how to address this issue is to read annotations in the operator. This could be either really specific, such as changing the translator logic to inject `env._managedServices = true` whenever it detects the presence of an `platform.stackrox.io/managed-services: true` annotation on the Central CR, or more general, via an annotation such as `platform.stackrox.io/inject-helm-values: '{"env": {"_managedServices": true}}'`.

I can see that this might seem like a bit of overengineering for the problem at hand, but I think it's not at all unlikely that we will see ourselves wanting to adjust the chart templates for managed services, beyond environment variables and beyond the knobs that the operator supports. But I want to hear your thoughts on that!

Reply:
I'd say that captures my thought process pretty much :-)
Thanks for the hints regarding the `GetCustomize` details!

I don't think the annotation way is over-engineered per-se, looking at other operators like the infamous ingress controllers this seems a typical solution to address this and it also seems better in the long-rung than doing the quick-way of using `._rox.customize.envVars`. 

I chose the `platform.stackrox.io/managed-services` annotation for that and the new helm setting `env.managedServices`. Now, we set `ROX_ENABLE_OPENSHIFT_AUTH / ROX_MANAGED_CENTRAL` only if the new setting is set.